Week5/app/src/main/java/com/oliverperez/blockbusterchallenge/ui/dashboard/MovieDashboardActivity.kt

```diff
@@ -0,0 +1,81 @@
+package com.oliverperez.blockbusterchallenge.ui.dashboard
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.Toast
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.oliverperez.blockbusterchallenge.ui.login.LoginActivity
+import com.oliverperez.blockbusterchallenge.ui.detail.MovieDetailActivity
+import com.oliverperez.blockbusterchallenge.model.LoginPrefs
+import com.oliverperez.blockbusterchallenge.model.MovieDataManager
+import com.oliverperez.blockbusterchallenge.model.Movie
+import com.oliverperez.blockbusterchallenge.R
+import com.oliverperez.blockbusterchallenge.viewmodel.MoviesViewModel
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MovieDashboardActivity : AppCompatActivity(), MovieGridAdapter.MovieClickListener {
+
+    private lateinit var moviesViewModel: MoviesViewModel
+    private lateinit var moviesRecyclerView: RecyclerView
+    private val dataManager = MovieDataManager()
+
+    companion object {
+        const val INTENT_MOVIE_KEY = "movie"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        moviesRecyclerView = movies_recycler_view
+        moviesRecyclerView.layoutManager = GridLayoutManager(this, 2)
+
+        if (!LoginPrefs.isUserLoggedIn()) {
+            startActivity(Intent(this, LoginActivity::class.java))
+        }
+        moviesViewModel = ViewModelProviders.of(this).get(MoviesViewModel::class.java)
+        moviesViewModel.clearAllMovies()
+        addDummyMovies()
+        moviesViewModel.getAllMovies().observe(this, Observer { movies ->
+            moviesRecyclerView.adapter =
+                MovieGridAdapter(
+                    movies,
+                    this
+                )
```

Comment:
This is not good practice. Now every time your data changes you will create a new adapter. That is completely unnecessary. You can create a lazy member variable which will hold the adapter and just initialize it once. 
Then whenever your data changes you should have a method in the adapter which you call to update the data. I commented that in the last PR if I recall correctly